commit 5fa8d04d9c86f343802c05bd3e11c6e733f01b63
Author: Keith Packard <keithp@keithp.com>
Date:   Tue Aug 25 18:33:04 2009 -0700

    Reload cursors as needed when setting new modes.
    
    Cursor images may need rotation, or positions updated when new modes
    are set. The server provides a convenience function,
    xf86_reload_cursors for precisely this purpose. Just call it after the
    new mode is set.
    
    Signed-off-by: Keith Packard <keithp@keithp.com>

diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index af5e56b..404f8b9 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -391,6 +391,8 @@ drmmode_set_mode_major(xf86CrtcPtr crtc, DisplayModePtr mode,
 
 	i830_set_max_gtt_map_size(pScrn);
 
+	if (pScrn->pScreen)
+		xf86_reload_cursors(pScrn->pScreen);
 done:
 	if (!ret) {
 		crtc->x = saved_x;
